storm-server/src/main/java/org/apache/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -256,7 +256,8 @@ private void scheduleTopology(TopologyDetails td, Cluster cluster, final User to
         if (forTest) {
             return this.evictedTopologies;
         } else {
-            throw new Exception("Topology eviction check is only provided for test purposes");
+            throw new UnsupportedOperationException(
```

Comment:
`UnsupportedOperationException` is a runtime exception. `throws Exception` is not needed in the signature. So is in the unit tests like testDefaultSchedulingPriorityStrategyNotEvicting

Reply:
Thanks for catching.